include/io/io_mesh_ascii.tcc

```diff
@@ -663,3 +737,25 @@ std::vector<std::tuple<mpm::Index, unsigned, double>>
   }
   return forces;
 }
+
+// Return array with math function entries
+template <unsigned Tdim>
+std::array<std::vector<double>, 2> mpm::IOMeshAscii<Tdim>::read_math_functions(
+    const std::string& math_file) {
+  // Initialise vector with 2 empty vectors
+  std::array<std::vector<double>, 2> xfx_values;
+
+  // Read from csv file
+  try {
+    io::CSVReader<2> in(math_file.c_str());
+    double x_value, fx_value;
+    while (in.read_row(x_value, fx_value)) {
+      xfx_values[0].push_back(x_value);
+      xfx_values[1].push_back(fx_value);
+    }
```

Comment:
The rest of the reader is in .txt, any specific consideration why we need csv for the function?

Reply:
If I remember correctly from yesterday, you felt that it is better to **not** require users to pre-process acceleration time histories (from NGA-WEST, etc.) from `.csv` -> `.txt`.

Reply:
Yes, but that's for acceleration, not function.

Reply:
The work by others is implemented like this:

```
      "acceleration_constraints": [
        {
          "nset_id" : 97,
          "math_function_id" : 0,
          "dir" : 0,
          "acceleration" : 1
        }
      ],
```

So the math function controlls the changes in the magntidue with time (i.e., time history) while `"acceleration"` becomes a scaling factor (e.g. `"acceleration": 1` is no scaling).

Reply:
So without including a `"math_function_id"`, the acceleration constraint will be constant (just like velocity or friction constraint)

Reply:
I am well aware, but we need to pre-process the NGA-West and split it into 2 or 3 CSV files right?

Reply:
Resolved as discussed in-person